parlai/crowdsourcing/utils/analysis.py

```diff
@@ -22,10 +22,34 @@ class AbstractResultsCompiler(ABC):
     def setup_args(cls):
         parser = argparse.ArgumentParser('Compile crowdsourcing results')
         parser.add_argument(
-            '--results-folders', type=str, help='Comma-separated list of result folders'
+            '--output-folder', type=str, help='Folder to save output files to'
         )
+        return parser
+
+    def __init__(self, opt: Dict[str, Any]):
+        self.output_folder = opt.get('output_folder')
+
+    @abstractmethod
+    def compile_results(self) -> pd.DataFrame:
```

Comment:
I'm not sure if DataFrame is the format of choice for all the data.
I think mine will be json for example.

Reply:
ah good to know - what will the format of your JSON be? The format of the output of `compile_results` is implicitly expected to be one utterance per row of the dataframe - I'm making a change and adding a comment about this

Reply:
For me it could be the context that is provided for that utterance as well. Actually I can have that as a df too, but I still rather having it as a JSON. Might be the case for some other people too.

Reply:
Yeah, the context also gets a row in the output dataframe format. What is it in particular that's helpful if you have a JSON format and not a df format? (Presumably you have a structured output that's hard to represent as a dataframe?)

Reply:
Not really, I just think that look better and more portable, and readable.

Reply:
Lost about `Union[pd.DataFrame, List]`?

Reply:
Hmm, if we want to support multiple output types, though, `Union[pd.DataFrame, list]` might still not be permissive enough, because perhaps someone might want to save their output as a single dict instead of a list of dicts, for instance.

My preference would still be for us to have a single output type if possible, because that'll reduce complexity in the future of having to support multiple ways of representing the same thing. That personally matters to me more than whether the return type is a dataframe or JSON format, for instance. I'll bring this up during the meeting next Tuesday, however, because I'm curious where the team as a whole stands on this

Reply:
Yeah, I agree with you. It is a good idea to decide on one to reduce the complexity and confusions.

Reply:
Meanwhile, I agree with you on keeping it DF for the time being.

Reply:
> Meanwhile, I agree with you on keeping it DF for the time being.

Okay cool - merging this in once CI checks pass